src/Models/Model.php

```diff
@@ -484,11 +484,11 @@ protected function purgeRelationship( string $key ): void {
 	 * @since 2.0.0
 	 *
 	 * @param string $key Relationship name.
-	 * @param Model|list<Model>|null $value The relationship value to cache.
+	 * @param mixed $value The relationship value to cache.
```

Comment:
Why this change?
I think the API should keep moving around `Model` instances (both in input and output), but internally (i.e. in the `Model::__costruct()` method parameters and in the database) can accept other values.
This change, like other going from `Model|Model[]` to `mixed`, complicate the use of the API, I think.

Reply:
I do not think the API should expose `$model->setCachedRelationship( 'post', 17 );`.
One should call `$model->setCachedRelationship( 'post', $model_for_post_17 );` and the API will work out what to store **internally**.

In general, I think the API change from `Model` to `mixed` breaks its ease of use. Now it can be anything and it goes back to being a leaky abstraction.